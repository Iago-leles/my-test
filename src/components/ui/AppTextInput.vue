<template>
  <div>
    <span class="text-subtitle1 text-grey-6 text-weight-bold">
      {{ label }}

      <span v-if="required" class="text-red-5">*</span>
    </span>

    <q-input v-model="model" v-bind="$attrs" />
  </div>
</template>

<script setup lang="ts">
import { ref } from 'vue'

const model = ref('')

defineProps<{
  label: string
  required: boolean
}>()
</script>
