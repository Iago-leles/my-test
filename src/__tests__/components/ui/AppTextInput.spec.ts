import { mount } from '@vue/test-utils'
import { expect, describe, it } from 'vitest'
import AppTextInput from '@/components/ui/AppTextInput.vue'

describe('AppTextInput', () => {
  it('should match snapshot', () => {
    const wrapper = mount(AppTextInput, {
      props: {
        label: 'Test Input',
        required: false,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should show required asterisk when required prop is true', () => {
    const wrapper = mount(AppTextInput, {
      props: {
        label: 'Test Input',
        required: true,
      },
    })

    expect(wrapper.find('.text-red-5').exists()).toBe(true)
  })

  it('should not show required asterisk when required prop is false', () => {
    const wrapper = mount(AppTextInput, {
      props: {
        label: 'Test Input',
        required: false,
      },
    })

    expect(wrapper.find('.text-red-5').exists()).toBe(false)
  })

  it('should update model value when input changes', async () => {
    const wrapper = mount(AppTextInput, {
      props: {
        label: 'Test Input',
        required: false,
      },
    })

    const input = wrapper.findComponent({ name: 'q-input' })
    await input.setValue('test value')

    expect((wrapper.vm as any).model).toBe('test value')
  })

  it('should display correct label', () => {
    const wrapper = mount(AppTextInput, {
      props: {
        label: 'Test Label',
        required: false,
      },
    })

    expect(wrapper.find('.text-subtitle1').text()).toContain('Test Label')
  })
})
