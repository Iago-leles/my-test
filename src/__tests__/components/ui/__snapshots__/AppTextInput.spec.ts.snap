// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppTextInput > should match snapshot 1`] = `
"<div><span class="text-subtitle1 text-grey-6 text-weight-bold">Test Input <!--v-if--></span><label class="q-field row no-wrap items-start q-field--standard q-input" for="f_598cbbff-b7c1-4d9b-abef-1d48a018bb92">
    <!---->
    <div class="q-field__inner relative-position col self-stretch">
      <div class="q-field__control relative-position row no-wrap" tabindex="-1">
        <div class="q-field__control-container col relative-position row no-wrap q-anchor--skip"><input class="q-field__native q-placeholder" tabindex="0" id="f_598cbbff-b7c1-4d9b-abef-1d48a018bb92" type="text" value="">
          <!---->
        </div>
      </div>
      <!---->
    </div>
    <!---->
  </label></div>"
`;
